src/betamax/cassette/cassette.py

```diff
@@ -67,6 +67,8 @@ def __init__(self, cassette_name, serialization_format, **kwargs):
         self.load_interactions()
         self.serializer.allow_serialization = self.is_recording()
 
+        dispatch_hooks("cassette_load", self)
```

Comment:
I wonder if it might be better to name this `before_load` and the other option `before_eject`.

I think `load` and `eject` might be poor names though. In theory, one wouldn't need to call `use_cassette` as a context manager so one could think of this as `post_start`, `post_stop`. In that case, I think I've talked myself into moving the hook dispatch to the start/stop methods here: https://github.com/betamaxpy/betamax/blob/d3f4bd89b2b56e11e0b03b6eb08c2b48727425fc/src/betamax/recorder.py#L121

Reply:
I'll use the names `after_start` and `after_stop` rather than the `post_` variants because those seem to match better with the `before_` names of the existing hooks. If you prefer the names to be `post_` I can of course change that.

Reply:
@sigmavirus24 At the time `start()` is called, is it possible for `self.betamax_adapter.cassette` to be `None`?

Reply:
So we don't enforce that `use_cassette` is called prior to `start` except in the case where users do:

```py
with recorder.use_cassette(...):
    # ...
```

Because we call `use_cassette` and return `self` which is a context manager which on `__enter__` calls `start()` and on `__exit__` calls `stop()`. So I think for >=98% of folks, they'll be in a good shape. Also I'm 99% certain that the test framework integrations I wrote to make this even easier do this correctly too. I think `cassette` should always be populated.